packages/client/src/Components/PaperEdits/PaperCutsPlayer/Player.js

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import VideoContext from 'videocontext';
+
+const getCanvas = (ref, width, height) => {
+  return (
+    <canvas
+      ref={ ref }
+      className='video-canvas'
+      width={ width }
+      height={ height }
+    >
+    </canvas>
+  );
+};
+
+class PaperCutsPlayer extends React.PureComponent {
+  constructor(props) {
+    const {
+      width = 640,
+      height = 360,
+      playlist = [],
+    } = props;
+
+    super(props);
+
+    this.playlist = playlist;
+    this.canvasRef = React.createRef();
+    this.canvas = getCanvas(this.canvasRef, width, height);
+  }
+
+  componentDidMount() {
+    this.videoContext = new VideoContext(this.canvasRef.current, () => console.log('oops'));
+    this.loadPlaylist();
+  }
+
+  loadPlaylist = () => {
+    this.playlist.forEach(({ type, offset, start, duration, src }) => {
+      const node = this.videoContext[type](src, offset);
+      node.startAt(start);
+      node.stopAt(start + duration);
```

Comment:
ok, didn't realise it was implemented like this under the hood using `duration` + `start` to get `end` time.
In which case is more likely our sequence of paper cuts instead of being
```js
{ type:'video', offset:0, start:0, duration:10, src:'http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ElephantsDream.mp4' },
```
is going to be
```js
{ type:'video', offset:0, start:0, end:10, src:'http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ElephantsDream.mp4' },
```

Similar to the [EDL composer sequence](https://github.com/OpenNewsLabs/autoEdit_2/tree/master/lib/edl_composer#usage)

---

However I'd say this is a change we can leave for a later stage tho, when the data structure for the papercuts sequence is a bit more flashed out.